src/inputValidationError.js

```diff
@@ -7,68 +7,76 @@
-            parsedError.push(this.parseAjvError(error, path, method));
-        }, this);
+const parseAjvError = function(error) {
+    return `${buildDataPath(error)} ${buildMessage(error)}`;
+};
 
-        return parsedError;
-    }
+const parseAjvErrors = function(errors) {
+    return errors.map(parseAjvError);
+};
 
-    parseAjvError(error, path, method) {
-        if (error.dataPath.startsWith('.header')) {
-            error.dataPath = error.dataPath.replace('.', '');
-            error.dataPath = error.dataPath.replace('[', '/');
-            error.dataPath = error.dataPath.replace(']', '');
-            error.dataPath = error.dataPath.replace('\'', '');
-            error.dataPath = error.dataPath.replace('\'', '');
-        }
+const buildMessage = function(error){
+    if (error.keyword === 'enum') {
+        return `${error.message} [${error.params.allowedValues.toString()}]`;
+    }
 
-        if (error.dataPath.startsWith('.path')) {
-            error.dataPath = error.dataPath.replace('.', '');
-            error.dataPath = error.dataPath.replace('.', '/');
-        }
+    if (error.validation) {
+        return error.errors.message;
+    }
 
-        if (error.dataPath.startsWith('.query')) {
-            error.dataPath = error.dataPath.replace('.', '');
-            error.dataPath = error.dataPath.replace('.', '/');
-        }
+    return error.message;
+};
 
-        if (error.dataPath.startsWith('.')) {
-            error.dataPath = error.dataPath.replace('.', 'body/');
-        }
+const buildDataPath = function(error) {
+    if (error.dataPath.startsWith('.header')) {
+        return error.dataPath
+            .replace('.', '')
+            .replace('[', '/')
+            .replace(']', '')
+            .replace('\'', '')
+            .replace('\'', '');
+    }
 
-        if (error.dataPath.startsWith('[')) {
-            error.dataPath = `body/${error.dataPath}`;
-        }
+    if (error.dataPath.startsWith('.path')) {
+        return error.dataPath
+            .replace('.', '')
+            .replace('.', '/');
+    }
 
-        if (error.dataPath === '') {
-            error.dataPath = 'body';
-        }
+    if (error.dataPath.startsWith('.query')) {
+        return error.dataPath
+            .replace('.', '')
+            .replace('.', '/');
+    }
 
-        if (error.keyword === 'enum') {
-            error.message += ` [${error.params.allowedValues.toString()}]`;
-        }
+    if (error.dataPath.startsWith('.')) {
+        return error.dataPath.replace('.', 'body/');
+    }
 
-        if (error.validation) {
-            error.message = error.errors.message;
-        }
+    if (error.dataPath.startsWith('[')) {
+        return `body/${error.dataPath}`;
+    }
 
-        return `${error.dataPath} ${error.message}`;
+    /* istanbul ignore next */
+    if (error.dataPath === '') {
+        return 'body';
     }
-}
 
-module.exports = InputValidationError;
\ No newline at end of file
+    /* istanbul ignore next */
```

Comment:
this ignore comment and the above one was added because any of the tests didn't get to the last line return undefined,but I rather stay this line of code because maybe there is scenario I didn't cover that cause ajv return error.datapath that doesn't match any of the if statements

Reply:
so please unit test for this scenario

Reply:
didn't check the scenario of error.datapath not defined because it can't be base on ajv error obj